src/runtime/hexagon/ops/conv2d.h

```diff
@@ -133,7 +155,48 @@ inline uintptr_t hwio_at(const DLTensor& f, int y, int x, int i, int o) {
  * @param width
  * @param depth
  */
-void blockize_hwc_16b(void* out, void* inp_flat, int height, int width, int depth);
+template <typename T, int block_height, int block_width, int block_depth>
+void blockize_hwc(void* out, void* inp_flat, int height, int width, int depth) {
```

Comment:
Would it make sense for `inp_flat`'s type to be `const void*` rather than `void*`?

This is probably a bit of a stylistic choice; I just figured I'd ask.

Reply:
I agree with both, I'll add the asserts and the `const void*` for the arguments, thanks.